kernel/src/sys/syscall.c

```diff
@@ -139,6 +139,26 @@ static void syscall_wm(registers_t* regs) {
                 wm_param_event_t* param = (wm_param_event_t*) regs->ecx;
                 wm_get_event(param->win_id, param->event);
         } break;
+        case WM_CMD_GET_POS: {
+            list_t* item = wm_get_window(regs->ecx);
+            if(item){
+                regs->eax = list_entry(item, wm_window_t)->pos;
+            } else {
+                printke("the given window id (%d) is unknown", regs->ecx);
+                regs->eax = -1;
+                break;    
+            }
+        } break;
```

Comment:
It actually is used once when setting up the other window type. But actually, this can be removed once the two windows are merged or something like that. Also, the position may be not that necessary for the applications themselves, so if you don't think both windows should share the same position, I can remove it

Reply:
I don't think `wm_window_t` and `window_t` will ever be merged, as they are really part of different applications: the window manager in the kernel (hopefully at some point moved to user space), and the UI library which defines its own concept of window.

I think it makes sense that a gui app can query the position of its window, so please keep it.  
However having the position stored in the UI-defined window type (see comment in "snow.h") doesn't make sense, because then the UI lib has to keep it in sync all the time (user moves window -> DRAGGED event callback -> syscall to get new pos) for no good reason the majority of the time (an app that needs to know its window position can make the call itself on demand).

Reply:
Woops, it seems that in my "freak go back" commit I accidentally removed the part where I set the ``point_t* window_t::pos`` to the syscall return. My fault again. In this case, I believe, we wouldn't need to update the position every event because it's a pointer to the actual structure, but if you wish so, I'm getting rid of it.

Reply:
That would not work, if it pointed to kernel memory, accessing it would crash the app (let me know if it doesn't, that would be a fundamental bug). The reverse could work, using process-level memory in the kernel and process (it's used to share window buffers, `ufb` in `wm_window_t`), but we don't want to complicate things that much.